controllers/genericPipeController.js

```diff
@@ -0,0 +1,90 @@
+"use strict"
+
+const { errObject } = require(`${__dirname}/../helpers/helper.js`)
+
+const genericPipeService = require(`${__dirname}/../services/genericPipeService`)
+
+exports.getGenericPipes = async (req, res, next) => {
+	try {
+		const pageNo = parseInt(req.query.pageNo) || 0
+		const docsPerPage = parseInt(req.query.docsPerPage) || 10
+		const response = await genericPipeService.getPagedGenericPipes(pageNo, docsPerPage)
+
+		res.status(200).json(response)
+	} catch (error) {
+		// Pass the error to the next middleware for centralized error handling
+		next(errObject("Error while fetching GenericPipes Data."))
+	}
+}
+
+exports.setGenericPipe = async (req, res, next) => {
+	try {
+		const genericPipeReq = req.body
+		// console.log(genericPipeReq)
+		const createdGenericPipeObj = await genericPipeService.createGenericPipe(genericPipeReq)
+		res.status(201).json({
+			status: "success",
+			message: "New generic pipe created successfuly.",
+			data: createdGenericPipeObj,
+		})
+	} catch (error) {
+		if (error.code !== 11000) next(error)
```

Comment:
same here
if (error.code !== 11000)
this check should not be in the controller but in the service function instead.

its fine if you add try catch there